.github/actions/cargo-build-macos-binary/action.yml

```diff
@@ -0,0 +1,70 @@
+name: "Build Quickwit binary for macOS"
+description: "Build React app and Rust binary for macOS with cargo build."
+inputs:
+  target:
+    description: "Target"
+    required: true
+  version:
+    description: "Binary version"
+    required: true
+  token:
+    description: "GitHub access token"
+    required: true
+runs:
+  using: "composite"
+  steps:
+    - run: echo "ASSET_FULL_NAME=quickwit-${{ inputs.version }}-${{ inputs.target }}" >> $GITHUB_ENV
+      shell: bash
+    - uses: actions/setup-node@v3
+      with:
+        node-version: 16
+        cache: "yarn"
+        cache-dependency-path: quickwit/quickwit-ui/yarn.lock
+    - run: make build-ui
+      shell: bash
+    - name: Install protoc
+      run: brew install protobuf
+      shell: bash
+    - uses: dtolnay/rust-toolchain@v1
```

Comment:
Wondering why you switched to this action for rust toolchain instead of the usual `actions-rs/toolchain@v1` .
Also, from the readme we need `dtolnay/rust-toolchain@master` since we are specifying the toolchain explicitly.

Reply:
> we need dtolnay/rust-toolchain@master since we are specifying the toolchain explicitly.

What do you mean? Is `dtolnay/rust-toolchain@v1` should not work?

Reply:
Using `dtolnay/rust-toolchain` seems like a good shot if you look at the repository. One is maintained, not the other. 
(example of issues with the unmaintained repos: https://github.com/actions-rs/toolchain/issues/219)

Reply:
dtolnay/rust-toolchain@v1 works but from the docs since we are using toolchain parameter, they recommend specifying `@master`. r
Regarding the fact this action is maintained, then we need to update all other places we use action-rs (in a separate PR)

Reply:
yes, I will keep using the action-rs/toolchain for now and will open an issue on this to migrate all workflows on this. This is not a priority :)